ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx

```diff
@@ -27,12 +84,277 @@ const translations = defineMessages({
   }
 });
 
-export default function AudiencesPanel() {
+interface AudiencesPanelUIProps {
+  audiencesResource: any;
+  onFormChange: Function;
+  saveProfile: Function;
+  setDefaults: Function;
+}
+
+export function AudiencesPanelUI(props: AudiencesPanelUIProps) {
+
+  const classes = useStyles({});
+  const {
+    audiencesResource,
+    onFormChange,
+    saveProfile,
+    setDefaults
+  } = props;
+  const audiencesData = audiencesResource.read();
+  const { config, profile } = audiencesData;
+
   return (
     <ToolPanel title={translations.audiencesPanel}>
-      <Typography component="h2" variant="subtitle1" style={{ padding: '10px' }}>
-        Audiences Panel
-      </Typography>
+      {
+        <>
+          <Grid className={classes.PanelMargin}>
+            {
+              Object.keys(config.fields).map((property: any, index: number) => (
+                <div key={index}>
+                  <AudiencesFormSection
+                    property={config.fields[property]}
+                    profileValue={profile[property] ? profile[property].key : undefined}
+                    profileTimezone={profile[`${property}_tz`] ? profile[`${property}_tz`].key : undefined}
+                    onFormChange={onFormChange}
+                  />
+                  <Divider className={classes.divider}/>
+                </div>
+              ))
+            }
+          </Grid>
+          <Grid className={classes.actionBTN}>
+            <Button variant="contained" onClick={() => setDefaults(config)}>
+              <FormattedMessage
+                id="audiencesPanel.defaults"
+                defaultMessage={`Defaults`}
+              />
+            </Button>
+            <Button variant="contained" color="primary" onClick={() => saveProfile()}>
+              <FormattedMessage
+                id="audiencesPanel.apply"
+                defaultMessage={`Apply`}
+              />
+            </Button>
+          </Grid>
+        </>
+      }
     </ToolPanel>
   );
 }
+
+export default function AudiencesPanel() {
+  const resource = useStateResourceSelection(
+    state => state.preview.audiencesPanel,
+    {
+      shouldRenew: (source, resource) => resource.complete,
+      shouldResolve: source => (!source.isFetching) && nnou(source.contentType) && nnou(source.model),
+      shouldReject: source => nnou(source.error),
+      errorSelector: source => source.error,
+      resultSelector: source => {
+        return {
```

Comment:
Keep the state names. Just do...
```js
resultSelector: source => pluckProps(source, 'contentType', 'model')
```

Reply:
now it's returning only contentType (per conversation)